src/api/java/com/minecolonies/api/compatibility/newstruct/BlueprintMapping.java

```diff
@@ -0,0 +1,173 @@
+package com.minecolonies.api.compatibility.newstruct;
+
+import java.util.*;
+
+/**
+ * The class that is responsible for the mapping of the old structurename/style to new style/structurename.
+ */
+public class BlueprintMapping
+{
+    /**
+     * Maps an old style name to a new style name.
+     */
+    private static Map<String, String> styleMapping = new HashMap<>();
+
+    /**
+     * Maps the old blueprint path to the new path. Query it with style:blueprintName and receive the pack specific path.
+     */
+    private static Map<String, String> pathMapping = new HashMap<>();
+
+    static
+    {
+        styleMapping.put("medievaloak", "Medieval Oak");
+        styleMapping.put("medievalspruce", "Medieval Spruce");
+        styleMapping.put("medievaldarkoak", "Medieval Dark Oak");
+        styleMapping.put("medievalbirch", "Medieval Birch");
+        styleMapping.put("caledonia", "Caledonia");
+        styleMapping.put("darkoak", "Dark Oak Treehouse");
+        styleMapping.put("fortress", "Fortress");
+        styleMapping.put("birch", "Urban Birch");
+        styleMapping.put("jungle", "Jungle Treehouse");
+        styleMapping.put("lostcity", "Lost Mesa City");
+        styleMapping.put("incan", "Incan");
+        styleMapping.put("nordic", "Nordic Spruce");
+        styleMapping.put("sandstone", "Desert Oasis");
+        styleMapping.put("spacewars", "Spacewars");
+        styleMapping.put("warped", "Warped Netherlands");
+        styleMapping.put("truedwarven", "Stalactite Caves");
+        styleMapping.put("stone", "Minecolonies Original");
+        styleMapping.put("acacia", "Urban Savanna");
+        styleMapping.put("wooden", "Minecolonies Original");
+        styleMapping.put("asian", "Pagoda");
```

Comment:
Still not really a fan of all this stuff.  Any way it could be put into the style pack metadata (json file or something) instead?  That way each new style can indicate what its legacy upgrade was, which would also support upgrades for out-of-jar styles.  You have to assume that these will not upgrade simultaneous with this landing, but a little later on, and that might rename/move some things that hopefully shouldn't break as a result.